### PR TITLE
perf: trtllm mla performance minor improvements

### PR DESCRIPTION
**Pull request by cicirori**
## Motivation

The current `trtllm_mla` backend has several minor inefficiencies and confusing parts:

1. Element-wise operations on `seq_lens` are repeatedly executed inside CUDA Graph replay, instead of being done once during metadata initialization.  
2. A redundant `torch.cat` operation (`k` + `k_rope`) exists in `draft_extend`, `draft`, and `verify` paths. This operation has been moved so it only happens during prefill.  
3. Some logic around sequence length metadata is confusing:
   - `metadata.max_seq_len_q = num_tokens_per_bs + 1` appears unnecessary.  
   - The handling of `spec_info.accept_length_cpu` likely needs a `+1` adjustment. (And why `accept_length=accept_length_cpu+1`?)

In earlier debugging, I encountered multiple `trtllm_mla`-related issues — such as low output performance with odd batch sizes and low output performance when `draft_extend` using cuda graph.  
This cleanup aims to improve both stability and runtime efficiency.

Preliminary benchmarks show a **~2% latency improvement** in decoding, and in faster speculative decoding scenarios, up to **4–5%** gain.

---

## Modifications

- Added `seq_lens_k` field to `TRTLLMMLADecodeMetadata` and initialize it properly.  
- Moved `seq_lens` arithmetic from CUDA graph replay into metadata initialization.  
- Refactored and unified logic in `init_forward_metadata_capture_cuda_graph`, `init_forward_metadata_replay_cuda_graph` and `init_forward_metadata`.  
- Removed redundant `metadata.max_seq_len_q = num_tokens_per_bs + 1`.  
- Clarified `spec_info.accept_length_cpu` handling and ensured consistent `+1` logic.  
- Deferred `k` / `k_rope` concatenation to prefill only.  
- Simplified `seq_lens` update expressions for clarity and efficiency.  

---

## Accuracy Tests 
gsm8k 10 run
image

image

## Benchmarking and Profiling 

```
SGLANG_ENABLE_SPEC_V2=1 \
python3 -m sglang.launch_server \
  --model nvidia/DeepSeek-V3-0324-FP4  \
  --trust-remote-code \
  --quantization modelopt_fp4 \
  --tp 4 \
  --attention-backend trtllm_mla \
  --moe-runner-backend flashinfer_trtllm \
  --speculative-algorithm EAGLE \
  --speculative-num-steps 3 \
  --speculative-eagle-topk 1 \
  --speculative-num-draft-tokens 4 \
  --model-loader-extra-config '{"enable_multithread_load": true, "num_threads": 8}' \
  --host 0.0.0.0 \
  --port 17345 \
  --enable-metrics \
  --enable-flashinfer-allreduce-fusion \
  --page-size 64

```

```
python3 -m sglang.test.send_one

pr:
+-------------+--------+------------+-----------------+
| Latency (s) | Tokens | Acc Length | Speed (token/s) |
+-------------+--------+------------+-----------------+
|    1.895    |  512   |   2.876    |     270.21      |
+-------------+--------+------------+-----------------+

main:
+-------------+--------+------------+-----------------+
| Latency (s) | Tokens | Acc Length | Speed (token/s) |
+-------------+--------+------------+-----------------+
|    1.939    |  512   |   2.876    |     264.11      |
+-------------+--------+------------+-----------------+
```

270.21/264.11 = 1.023

## Checklist 
- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit). 
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests). 
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations). 
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

## Notes / Discussion

If my interpretation of certain variables (especially `accept_length_cpu` vs. `accept_length`) is not accurate, please correct me.  
A well-documented explanation of the sequence length variables and their intended roles would make it much easier for contributors to understand and extend this backend in the future.

### CODE REVIEW

**Review by Fridge003 — APPROVED** [review]
Nice catch!

**Review by Qiaolin-Yu — APPROVED** [review]

**Comment by Fridge003** [comment]
https://github.com/sgl-project/sglang/actions/runs/18990381571/job/54246145107?pr=12435
